docs/docs/rendering-elements.md

```diff
@@ -3,6 +3,79 @@ id: rendering-elements
 title: Rendering Elements
 permalink: docs/rendering-elements.html
 prev: introducing-jsx.html
+next: components-and-props.html
 ---
 
-TODO
+Elements are the smallest building blocks of React apps.
+
+An element describes what you want to see on the screen:
+
+```js
+const element = <h1>Hello, world</h1>;
+```
+
+Unlike browser DOM elements, React elements are plain objects, and are cheap to create. React DOM takes care of updating the DOM to match the React elements.
+
+## Rendering an Element into the DOM
+
+Let's say there is a `<div>` somewhere in your HTML file:
+
+```html
+<div id="root"></div>
+```
+
+We call this a "container" DOM node because the React app is contained by it. Everything inside it will be managed by React DOM.
+
+Applications built with just React usually have a single container node. If you are integrating React into an existing app, you may have as many isolated container nodes as you like.
+
+To render a React element into a DOM container, pass both to `ReactDOM.render()`:
+
+```js{3}
+const container = document.getElementById('root');
+const element = <h1>Hello, world</h1>;
+ReactDOM.render(element, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/rrpgNB?editors=1010)
+
+It displays "Hello World" on the page.
+
+## Updating the Rendered Element
+
+React elements are [immutable](https://en.wikipedia.org/wiki/Immutable_object). Once you create an element, you can't change it.
+
+When we want the UI to update, we create a new element, and pass it to `ReactDOM.render()`. It updates the DOM to match the newly passed element.
+
+Consider this ticking clock example:
+
+```js{10}
+const container = document.getElementById('root');
+
+function tick() {
+  const element = (
+    <div>
+      <h1>Hello, world!</h1>
+      <h2>It is {new Date().toLocaleTimeString()}.</h2>
+    </div>
+  );
+  ReactDOM.render(element, container);
```

Comment:
This seems like it isn't a good pattern to advise - would you ever want to update your React app that way? If someone cut and pasted this pattern into their app it would probably end up being a mess.

Maybe the whole concept of "How UIs update" should go later, when state is introduced. And this page could demonstrate rendering an element, and it could also demonstrate rendering a tree of elements, but not show how `render` can be used many times into the same DOM node because that doesn't seem like something we want to advocate.